consider adding localization to your original project

pparedes1 opened the pull request:
Hi Felix, when I originally forked your project I simply created an English translation, but after playing around with i18next, I added localization with English and German so that both versions could live in one repo. If these changes are in line with how you would do them, consider incorporating into your repo. I've added a few more apps, though not sure if we need to add localization to the apps, in case European URLs for things like amazon or plex are different. Any feedback is welcome (including how crappy my code is :)) but even if you don't incorporate, thanks again for letting me play around with your nifty app.

Pedro P.

pparedes1 committed:
converted to english, updated dependencies, and revised README

pparedes1 committed:
updated intentSchema JSON using Amazon's new interactive builder and updated README

pparedes1 committed:
added localization using i18next, added English and German translations

pparedes1 committed:
added additional standard applications like webbrowser, notifications, gallery, etc to lookup.js

pparedes1 committed:
dependency fix, updated default constants for deployments

pparedes1 committed:
fixed issue when disconnecting, added some debugging to console with LGDEBUG environment variable

voydz commented:
Hey @pparedes1 ,

thank you for your work, much appreciated! I totally like the idea of a multilingual approach. But before we can merge this we have to go through the meta stuff. I see you edited i.e. the Readme and some linting options. 

Especially the readme does not fit in the repo (along the lines: "this is a fork of ..." 😄) Would you mind looking over those things @pparedes1 ? I will doublecheck and make final adjustments myself before merging.

Again, thank you for your time you invest in this project.
best regards

pparedes1 commented:
Hey @voydz that makes perfect sense, didn’t think of it! I can update in a future commit and see if you want to incorporate. I’m sure you’ll want to squash a few of my commits either way, since I changed direction a few times on the whole language thing.

I could start placing changes on a feature/develop branch if you’d like (didn’t know if you were interested in merging so I assumed it would be a one developer repo and did a few forced pushes to the master branch but I won’t anymore 😀). Let me know what makes sense for you if you prefer only fast forward merges, etc.

Also see if you are ok with the gulp file or prefer a different location for those assets for translation. I updated to V4, not a big deal but tried to keep your original flow. As for the linting I changed the linebreaks as I was working on a windows pc and hated changing on my side, not sure how critical of an error or if this should be a warning...

The dubugging flag is inelegant at this point as I was just playing around, but we might consider some type of logging function. Finally the last change I made was a callhome function to keep this app alive on Heroku (it shuts down the server if it doesn't get a call after 20 mins so I added a variable to call itself), clearly not pretty.

pparedes1 committed:
Updated readme for clarity

voydz commented:
@pparedes1 Thank you very much, I merged your PR in a new "develop" branch. This branch will contain staged changes before merging into master. 

I would be very glad if someone is able to test all changes prior to merging into master. I can not do this myself right now, as I am on vacation. 😎 A short feedback would be nice.
